freeipa/src/main/java/com/sequenceiq/freeipa/entity/Stack.java

```diff
@@ -126,6 +126,10 @@ public class Stack implements AccountAwareResource, OrchestratorAware, IdAware {
     @SecretValue
     private Secret databusCredential = Secret.EMPTY;
 
+    @Convert(converter = SecretToString.class)
+    @SecretValue
+    private Secret monitoringCredential = Secret.EMPTY;
```

Comment:
maybe these should be in a separate table. I don't think every time we fetch the stack all these users are necessary. could you investigate it and file a followup to move the users out?

Reply:
not sure im agreeing with you or not...you rightish...note that we are following the same that how it was done with the secrets in Cluster.java, but there is no Cluster object for freeipa, so it's in the Stack

Reply:
you can't grow a table indefinitely. these are not used every time we fetch the stack. maybe in case of cluster these also should be moved out, as the cluster table is also overwhelmed. I know it's very easy just to add a new field/column, but that doesn't mean it's the right way.

Reply:
I created a separate ticket for this https://jira.cloudera.com/browse/CB-18229. We can migrate these secrets into  a separate table in two phases. In one we store them into a different table as well. Next we migrate all of them to the new table. Since this would be a bigger effort I created a new ticket for it.